Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.kt

```diff
@@ -0,0 +1,151 @@
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class TagDialogFragment(private val tag: Tag) : AppCompatDialogFragment(), OnShowListener {
+    private val viewModel: TagDialogViewModel by viewModels()
+
+    private var _dialogEditTag: AlertDialog? = null
+    private val dialogEditTag get() = _dialogEditTag!!
+
+    private var _binding: EditTagBinding? = null
+    private val binding  get() = _binding!!
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        viewModel.close()
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        _binding = EditTagBinding.inflate(LayoutInflater.from(context))
+        return buildDialog()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+        _dialogEditTag = null
+    }
+
+    private fun buildDialog(): Dialog {
+        val context: Context = ContextThemeWrapper(requireContext(), R.style.Dialog)
+        _dialogEditTag = AlertDialog.Builder(context)
+            .setView(binding.root)
+            .setTitle(R.string.rename_tag)
+            .setNegativeButton(R.string.cancel, null)
+            .setPositiveButton(R.string.save, null)
+            .create()
+
+        dialogEditTag.setOnShowListener(this)
+        binding.inputTagName.editText?.doAfterTextChanged { s -> viewModel.updateUiState(s.toString()) }
+
+        return dialogEditTag
+    }
+
+    private fun setObservers() {
+        viewModel.uiState.observe(this, { (_, _, errorMsg) ->
+            if (errorMsg != null) {
+                binding.inputTagName.error = getString(errorMsg)
+                dialogEditTag.getButton(DialogInterface.BUTTON_POSITIVE).isEnabled = false
+            } else {
+                binding.inputTagName.error = null
+                dialogEditTag.getButton(DialogInterface.BUTTON_POSITIVE).isEnabled = true
+            }
+        })
+
+        viewModel.event.observe(this, { event: TagDialogEvent ->
+            when (event) {
+                CloseEvent, FinishEvent -> dismiss()
+                ShowErrorEvent -> {
+                    val context = requireContext()
+                    DialogUtils.showDialogWithEmail(context, context.getString(R.string.rename_tag_message))
+                }
+                is ConflictEvent -> {
+                    val (canonicalTagName, oldTagName) = event
+                    showDialogErrorConflict(canonicalTagName, oldTagName)
+                }
+            }
+        })
+    }
+
+    override fun onShow(dialog: DialogInterface) {
+        setObservers()
+        startUiState()
+        showDialogRenameTag()
+    }
+
+    private fun startUiState() {
+        viewModel.start(tag)
+        binding.inputTagName.editText?.setText(tag.name)
+    }
+
+    /**
+     * Change the dialog layout to show a conflict message in which the user can merge two tags or cancel the merge
```

Comment:
Minor (🔍): Consider ending the KDoc sentence with a dot (`.`).

PS: Same apply for the showDialogRenameTag() function below.

Reply:
Good catch! Fixed in b41944abc5f2a0b8a0ac1c4fca8e9c567e8010d9.